### PR TITLE
SDK batching/revamp 2.1: `clock` example for Rust

### PR DESCRIPTION
**Pull request by teh-cmc**
What the title says.

On top of #1983

### CODE REVIEW

File: examples/rust/clock/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+rust-version.workspace = true
+license.workspace = true
+publish = false
+
+[dependencies]
+rerun = { workspace = true, features = ["web_viewer"] }
+
+anyhow.workspace = true
+clap = { workspace = true, features = ["derive"] }
+glam.workspace = true
```
Comment (teh-cmc) [inline examples/rust/clock/Cargo.toml:14]: no workspace refs

File: examples/rust/clock/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    steps: usize,
+}
+
+fn run(rec_stream: &RecordingStream, args: &Args) -> anyhow::Result<()> {
+    const LENGTH_S: f32 = 20.0;
+    const LENGTH_M: f32 = 10.0;
+    const LENGTH_H: f32 = 4.0;
+    const WIDTH_S: f32 = 0.25;
+    const WIDTH_M: f32 = 0.5;
+    const WIDTH_H: f32 = 0.7;
```
Comment (teh-cmc) [inline examples/rust/clock/src/main.rs:33]: fix the python example btw

**Review by Wumpf — APPROVED** [review]
makes sense, works as documented. Nice to have more rust examples for testing

File: examples/rust/clock/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        Ok(())
+    }
+
+    for step in 0..args.steps {
+        #[rustfmt::skip]
+        log_hand(rec_stream, "seconds", step, (step % 60) as f32 / 60.0, LENGTH_S, WIDTH_S, 0)?;
+        #[rustfmt::skip]
+        log_hand(rec_stream, "minutes", step, (step % 3600) as f32 / 3600.0, LENGTH_M, WIDTH_M, 128)?;
+        #[rustfmt::skip]
+        log_hand(rec_stream, "hours", step, (step % 43200) as f32 / 43200.0, LENGTH_H, WIDTH_H, 255)?;
```
[the remark this answers is not in the document]
Comment (teh-cmc) [reply examples/rust/clock/src/main.rs:99]: Not on expressions, it's been nightly-only since the dawn of time... :unamused:

**Review by Wumpf — CHANGES_REQUESTED** [review]
looks like you need to update cargo.lock again, noticed only when switching to a different branch now:

```
diff --git a/Cargo.lock b/Cargo.lock
index 304bd0417..2124b977e 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -773,7 +773,7 @@ name = "clock"
 version = "0.6.0-alpha.0"
 dependencies = [
  "anyhow",
- "clap 4.1.4",
+ "clap 4.1.14",
  "glam",
  "rerun",
 ]
```

**Comment by Wumpf** [comment]
ah too late, uhoh. Guess this might give @jprochazk the main ci failure test 😄

**Comment by teh-cmc** [comment]
Which will be in about ten minutes thanks to #2026 :eyes:

**Comment by jprochazk** [comment]
Doesn't seem to have caused a failure 😢

**Comment by teh-cmc** [comment]
> Doesn't seem to have caused a failure cry

Yeah I'm not surprised: outdated lock files are annoying because they screw with everyone's local copy, but I don't remember we ever configured anything to detect these situations on the CI.

**Pull request merged**